pkg/cmd/auth/login/login.go

```diff
@@ -71,6 +72,12 @@ func NewCmdLogin(f *cmdutil.Factory, runF func(*LoginOptions) error) *cobra.Comm
 				if opts.Hostname == "" {
 					opts.Hostname = ghinstance.Default()
 				}
+			} else {
+				if !opts.IO.CanPrompt() {
+					// TODO this error message kind of sucks since the prompting might be disabled from being
+					// in a nontty or from the config setting. Is it enough info?
+					return &cmdutil.FlagError{Err: errors.New("--with-token required when prompts disabled")}
```

Comment:
I think we should avoid saying “when prompts disabled” because `CanPrompt` can return false even if the user hasn't explicitly disabled prompts. Maybe `--with-token required when not running in interactive mode`?